### ISSUE TITLE
New Analysis Idea - Most Central Feature

### ISSUE THREAD

**Issue by stuartlynn** [CartoDB/crankshaft#107]
## Description 
Analysis to find the most central feature of a dataset with the option of doing so in categories. 

## Inputs 
-Layer with points, polygons or lines 
-column to use for weighting
-column to use for categories 

## Outputs 

- A table with two columns, the most central feature id and the category it is the most central of. 

## Usecases 

- Centralized placement of resources at existing structures.

**Comment by AbelVM** [comment]
Maybe this can be achieved by a simple *Camshaft* node:

```sql
WITH 
a AS ( 
    {{=it.layer_query}} 
),
b AS (
    SELECT
    {{=it.category}} as category 
    avg( {{=it.weight}} ) as center
    FROM a
    GROUP BY {{=it.category}}
),
c AS (
    SELECT
        a.cartodb_id,
        b.category,
        (a.{=it.weight}} - b.center) as dev
    FROM a LEFT JOIN b 
    ON a.{{=it.category}} = b.category
),
d AS(
    SELECT
        category,
        min(dev) as result_w
    FROM c
    GROUP BY category
)
SELECT
    c.cartodb_id,
    c.category
FROM d LEFT JOIN c
ON d.result_w = c.dev
```

Or do we need it as a function available DB level?

**Comment by stuartlynn** [comment]
Yeah this is a super simple one. I don't think it hurts to have it at the DB level. I think where possible we should try to wrap analysis in a function so we can call them through the API as well as the builder interface

**Comment by AbelVM** [comment]
Ok, almost same code

**Comment by ohasselblad** [comment]
Yeah, I've been wondering about what we offer in crankshaft vs what we offer in camshaft. Should the API we offer for crankshaft be in parity with what's offered in the Builder? Because right now there's an 'arbitrary' collection in here and it might confuse users who want to do a specific analysis from the Engine.

**Comment by ohasselblad** [comment]
Which I guess is a larger discussion of an Analysis API...
